tensorboard/webapp/core/effects/core_effects.ts

```diff
@@ -172,11 +171,54 @@ export class CoreEffects {
         })
       );
 
-      const runsReload$ = this.onDataReload$.pipe(
+      const runsReload$ = this.onDashboardLoad$.pipe(
+        map(([, routeKind]) => routeKind),
+        switchMap((routeKind) => {
+          if (routeKind !== RouteKind.COMPARE_EXPERIMENT) {
+            return of([]);
+          }
+
+          return this.store.select(getExperimentIdToAliasMap).pipe(
+            distinctUntilChanged((beforeAliasDict, afterAliasDict) => {
```

Comment:
Can you add a (possibly brief) comment explaining what the code is trying to achieve here? Or move into its own function whose name describes the code?